pkg/ingester/ingester.go

```diff
@@ -534,7 +544,15 @@ func (i *Ingester) startingForFlusher(ctx context.Context) error {
 
 func (i *Ingester) starting(ctx context.Context) (err error) {
 	defer func() {
-		if err != nil {
+		if err == nil {
+			if i.cfg.CircuitBreakerConfig.InitialDelay == 0 {
```

Comment:
This logic can be at the end of `starting` method, if there are no errors. There is no need to put it into `defer`.

Reply:
I have moved the logic at the end of `starting`.